ZenodoExportPlugin.php

```diff
@@ -134,56 +159,107 @@ public function getSettingsFormClassName(): string
+        }
 
-        try {
-            $response = $httpClient->request(
-                'POST',
-                $url,
-                [
-                    'headers' => $headers,
-                    'json' => json_decode($jsonString),
-                ]
-            );
-        } catch (GuzzleException | Exception $e) {
-            return [['plugins.importexport.zenodo.register.error.mdsError', $e->getMessage()]];
+        if ($isUpdate && $isPublished) {
+            // try to re-publish the updated record
+            $republishUpdatedRecord = $this->publishZenodoRecord($object, $zenodoRecId, $recordsApiUrl, $apiKey);
+            if (is_array($republishUpdatedRecord)) {
+                return $republishUpdatedRecord;
+            }
         }
 
-        $responseBody = json_decode($response->getBody());
-        $zenodoRecId = $responseBody->id;
+        // note: can't update files on published records
+        if (!$isUpdate) { // @todo disable file deposit for updates as it will fail if the filename exists already - maybe we should delete all files or the record first in this case
+            $filesDeposit = $this->depositFiles($object, $recordsApiUrl, $apiKey, $zenodoRecId);
+            if (is_array($filesDeposit)) {
+                // File deposit has failed, try to delete the record in Zenodo then return the error
+                $this->deleteRecord($object, $zenodoRecId, $recordsApiUrl, $apiKey);
+                return $filesDeposit;
+            }
+        }
 
-        if (($status = $response->getStatusCode()) != ZENODO_API_DEPOSIT_CREATED) {
-            return [['plugins.importexport.zenodo.register.error.mdsError', $status . ' - ' . $response->getBody()]];
+        // Submit the record to a community
+        $communityId = $this->getCommunityId($context);
+        if ($communityId) {
+            if ($review = $this->createReview($zenodoRecId, $communityId, $recordsApiUrl, $apiKey)) {
+                // @todo depending on the community permissions, this can automatically publish the record - make sure this is clear to the user.
+                $requestId = $this->submitReview($zenodoRecId, $zenodoApiUrl, $apiKey);
+                $autoPublishCommunity = $this->automaticPublishingCommunity($context);
+                if (is_array($requestId)) {
+                    return $requestId;
+                } elseif ($autoPublishCommunity && $requestId) {
+                    // @todo handle errors - if this fails, delete the review request?
+                    $reviewAccepted = $this->acceptReview($requestId, $recordsApiUrl, $apiKey);
+                }
+            } elseif (is_array($review)) {
+                return $review;
+            }
         }
 
-        // if the submission has files
-        $this->depositFiles($object, $zenodoRecId, $url, $apiKey);
+        // @todo consider how to handle this alongside auto publishing in community
+        // (e.g. record may be published already)
```

Comment:
Can we check here if already published (through community)? At this place it can be only published through community, so could we maybe put this part of the code in front of the community insertion above?